mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -35,6 +36,7 @@ abstract class BaseMvRxViewModel<S : MvRxState>(
     private val tag by lazy { javaClass.simpleName }
     private val disposables = CompositeDisposable()
     private lateinit var mutableStateChecker: MutableStateChecker<S>
+    private val lastDeliveredStates = ConcurrentHashMap<String, Any>()
```

Comment:
I don't think there is any check that the same id isn't used twice in separate subscriptions. The potential bug where different subscriptions with the same properties leads to collision worries me, especially since it doesn't seem like it would be an obvious issue.

Could we store a map of Id to Disposable here, and it is only valid to register the same id if the disposable is disposed?

Reply:
Good idea! I'll add this, but this makes it very important for us to fix the duplicate subscriptions if you subscribe outside of onCreate.

I have a fix in progress for that.